### PR TITLE
BLD: Fix minimum Python version in setup.py

### PR DESCRIPTION
**Pull request by astrojuanlu**
See https://github.com/numpy/numpy/issues/12480 for discussion.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -26,15 +26,9 @@
 import subprocess
 import textwrap
 
-
```
Comment (Juanlu001) [inline setup.py]: In principle this is not necessary, assuming the user is running pip>=9.0.0.

File: setup.py [hunk 2]
```diff
@@ -26,15 +26,9 @@
...
-if sys.version_info[:2] < (3, 5):
```
Comment (Juanlu001) [inline setup.py]: In principle this is not necessary, assuming the user is running pip>=9.0.0.

File: setup.py [hunk 3]
```diff
@@ -26,15 +26,9 @@
...
-    raise RuntimeError("Python version >= 3.5 required.")
-
 import builtins
 
 
-python_requires = '>=3.5'
```
Comment (Juanlu001) [inline setup.py]: This variable was not being used. An alternative is to do `python_requires=python_requires` below.

File: setup.py (hunk 2 above)
Comment (rgommers) [reply setup.py]: not everyone is using `pip` to install ....

we should keep this, clear error message helps

File: setup.py (hunk 3 above)
Comment (rgommers) [reply setup.py]: deleting this seems right, no need for this variable

**Comment by rgommers** [comment]
Thanks @Juanlu001. If you undo the removal of the explicit version check I'll merge this.

**Comment by Juanlu001** [comment]
Thanks for the quick review @rgommers! I restored the error message.

File: setup.py (hunk 2 above)
Comment (Juanlu001) [reply setup.py]: Changed

**Pull request merged**